app/services/post_alerter.rb

```diff
@@ -180,8 +180,17 @@ def group_watchers(topic)
 
   def tag_watchers(topic)
     topic.tag_users
-      .where(notification_level: TagUser.notification_levels[:watching_first_post])
-      .pluck(:user_id)
+      .joins("LEFT OUTER JOIN tag_group_memberships ON tag_users.tag_id = tag_group_memberships.tag_id")
+      .joins("LEFT OUTER JOIN tag_group_permissions ON tag_group_memberships.tag_group_id = tag_group_permissions.tag_group_id")
+      .joins("LEFT OUTER JOIN group_users on group_users.user_id = tag_users.user_id")
+      .where("(tag_group_permissions.group_id IS NULL
+               OR tag_group_permissions.group_id = group_users.group_id
+               OR group_users.group_id = ?)
+              AND tag_users.notification_level = ?",
+              Group::AUTO_GROUPS[:staff],
+              TagUser.notification_levels[:watching_first_post],
```

Comment:
what about :watching? do we need that as well for the test?

Reply:
and even tracking, may be subtle, but you would start tracking the topic and know by association that it has the tag

Reply:
For this particular `tag_watchers` method we only need to consider `:watching_first_post` as it's used for `notify_first_post_watchers`. The other notification levels are covered elsewhere.

https://github.com/discourse/discourse/blob/c50434be63f4466a44c17f6a339f203d5ce21d17/app/services/post_alerter.rb#L162-L169

Reply:
I'll cover them in this PR as well. Converting to draft.